cmd/ci-secret-generator/main_test.go

```diff
@@ -0,0 +1,391 @@
+						},
+					},
+				},
+			},
+			expected: []bitWardenItem{
+				{
+					ItemName: "Item1",
+					Fields: []fieldGenerator{
+						{
+							Name: "Field1",
+							Cmd:  "echo -n Field1",
+						},
+					},
+					Attachments: []fieldGenerator{
+						{
+							Name: "Attachment1",
+							Cmd:  "echo -n Attachment1",
+						},
+					},
+					Attribute: fieldGenerator{
+						Name: "password1",
+						Cmd:  "echo -n password1",
+					},
+					Notes: "Note1",
+					Params: map[string][]string{
+						"FieldNum": {
+							"1",
+							"2",
+						},
+					},
+				},
+				{
+					ItemName: "Item2",
+					Fields: []fieldGenerator{
+						{
+							Name: "Field2",
+							Cmd:  "echo -n Field2",
+						},
+					},
+					Attachments: []fieldGenerator{
+						{
+							Name: "Attachment2",
+							Cmd:  "echo -n Attachment2",
+						},
+					},
+					Attribute: fieldGenerator{
+						Name: "password2",
+						Cmd:  "echo -n password2",
+					},
+					Notes: "Note2",
+					Params: map[string][]string{
+						"FieldNum": {
+							"1",
+							"2",
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "Two parameters with multiple values",
+			inputBwItems: []bitWardenItem{
+				{
+					ItemName: "Item$(FieldNum)$(Env)",
+					Fields: []fieldGenerator{
+						{
+							Name: "Field$(FieldNum)$(Env)",
+							Cmd:  "echo -n Field$(FieldNum)$(Env)",
+						},
+					},
+					Attachments: []fieldGenerator{
+						{
+							Name: "Attachment$(FieldNum)$(Env)",
+							Cmd:  "echo -n Attachment$(FieldNum)$(Env)",
+						},
+					},
+					Attribute: fieldGenerator{
+						Name: "password$(FieldNum)$(Env)",
+						Cmd:  "echo -n password$(FieldNum)$(Env)",
+					},
+					Notes: "Note$(FieldNum)$(Env)",
+					Params: map[string][]string{
+						"FieldNum": {
+							"1",
+							"2",
+						},
+						"Env": {
+							"dev",
+							"prod",
+						},
+					},
+				},
+			},
+			expected: []bitWardenItem{
+				{
+					ItemName: "Item1dev",
+					Fields: []fieldGenerator{
+						{
+							Name: "Field1dev",
+							Cmd:  "echo -n Field1dev",
+						},
+					},
+					Attachments: []fieldGenerator{
+						{
+							Name: "Attachment1dev",
+							Cmd:  "echo -n Attachment1dev",
+						},
+					},
+					Attribute: fieldGenerator{
+						Name: "password1dev",
+						Cmd:  "echo -n password1dev",
+					},
+					Notes: "Note1dev",
+					Params: map[string][]string{
+						"FieldNum": {
+							"1",
+							"2",
+						},
+						"Env": {
+							"dev",
+							"prod",
+						},
+					},
+				},
+				{
+					ItemName: "Item1prod",
+					Fields: []fieldGenerator{
+						{
+							Name: "Field1prod",
+							Cmd:  "echo -n Field1prod",
+						},
+					},
+					Attachments: []fieldGenerator{
+						{
+							Name: "Attachment1prod",
+							Cmd:  "echo -n Attachment1prod",
+						},
+					},
+					Attribute: fieldGenerator{
+						Name: "password1prod",
+						Cmd:  "echo -n password1prod",
+					},
+					Notes: "Note1prod",
+					Params: map[string][]string{
```

Comment:
Seems weird to have these on the output type since they've been consumed in the resolution

Reply:
True, but to remove this alone I would need to start copying the terms explicitly instead of just using a deep copy function. I didn't think that removing the params field is worth moving stuff over by hand.